gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -476,33 +475,26 @@ impl GatewayLightning for ClnRpcService {
 
     async fn route_htlcs(
         &self,
-        request: tonic::Request<tonic::Streaming<InterceptHtlcResponse>>,
+        _: tonic::Request<EmptyRequest>,
     ) -> Result<tonic::Response<Self::RouteHtlcsStream>, Status> {
-        let mut stream = request.into_inner();
-
         // First create new channel that we will use to send responses back to gatewayd
         let (gatewayd_sender, gatewayd_receiver) =
             mpsc::channel::<Result<InterceptHtlcRequest, Status>>(100);
 
         let mut sender = self.interceptor.sender.lock().await;
         *sender = Some(gatewayd_sender.clone());
 
-        // Spawn new thread that listens for events from the input stream
-        let interceptors = self.interceptor.clone();
-        tokio::spawn(async move {
-            while let Some(res) = stream.next().await {
-                if let Ok(complete_request) = res {
-                    let _ = Self::complete_htlc(complete_request, interceptors.clone())
-                        .await
-                        .map_err(|e| {
-                            error!("CLN extension failed to complete HTLC: {:?}", e);
-                        });
-                }
-            }
-        });
```

Comment:
I assume that deleted code was concerned with processing the HTLC completion messages streamed to us by the client?

Reply:
Yep, exactly. A little more context here: https://github.com/fedimint/fedimint/pull/2692/files#r1262890249